velox/connectors/hive/iceberg/tests/IcebergReadTest.cpp

```diff
@@ -170,66 +171,76 @@ class HiveIcebergTest : public HiveConnectorTestBase {
-        writeDataFiles(rowGroupSizesForFiles);
-    std::unordered_map<
-        std::string,
-        std::pair<int64_t, std::shared_ptr<TempFilePath>>>
-        deleteFilePaths = writePositionDeleteFiles(
-            deleteFilesForBaseDatafiles, dataFilePaths);
-
-    std::vector<std::shared_ptr<ConnectorSplit>> splits;
-
-    for (const auto& dataFile : dataFilePaths) {
-      std::string baseFileName = dataFile.first;
-      std::string baseFilePath = dataFile.second->getPath();
-
-      std::vector<IcebergDeleteFile> deleteFiles;
-
-      for (auto const& deleteFile : deleteFilesForBaseDatafiles) {
-        std::string deleteFileName = deleteFile.first;
-        std::multimap<std::string, std::vector<int64_t>> deleteFileContent =
-            deleteFile.second;
-
-        if (deleteFileContent.count(baseFileName) != 0) {
-          // If this delete file contains rows for the target base file, then
-          // add it to the split
-          auto deleteFilePath =
-              deleteFilePaths[deleteFileName].second->getPath();
-          IcebergDeleteFile deleteFile(
-              FileContent::kPositionalDeletes,
-              deleteFilePath,
-              fileFomat_,
-              deleteFilePaths[deleteFileName].first,
-              testing::internal::GetFileSize(
-                  std::fopen(deleteFilePath.c_str(), "r")));
-          deleteFiles.push_back(deleteFile);
+    std::vector<dwio::common::FileFormat> fileFormats = {
+        dwio::common::FileFormat::DWRF};
+    if (hasWriterFactory(dwio::common::FileFormat::PARQUET)) {
+      fileFormats.push_back(dwio::common::FileFormat::PARQUET);
+    }
+
+    for (dwio::common::FileFormat fileFormat : fileFormats) {
+      // Keep the reference to the deleteFilePath, otherwise the corresponding
+      // file will be deleted.
+      std::map<std::string, std::shared_ptr<TempFilePath>> dataFilePaths =
+          writeDataFiles(rowGroupSizesForFiles, fileFormat);
+      std::unordered_map<
+          std::string,
+          std::pair<int64_t, std::shared_ptr<TempFilePath>>>
+          deleteFilePaths = writePositionDeleteFiles(
+              deleteFilesForBaseDatafiles, dataFilePaths, fileFormat);
+
+      std::vector<std::shared_ptr<ConnectorSplit>> splits;
+
+      for (const auto& dataFile : dataFilePaths) {
+        std::string baseFileName = dataFile.first;
+        std::string baseFilePath = dataFile.second->getPath();
+
+        std::vector<IcebergDeleteFile> deleteFiles;
+
+        for (auto const& deleteFile : deleteFilesForBaseDatafiles) {
+          std::string deleteFileName = deleteFile.first;
+          std::multimap<std::string, std::vector<int64_t>> deleteFileContent =
+              deleteFile.second;
+
+          if (deleteFileContent.count(baseFileName) != 0) {
+            // If this delete file contains rows for the target base file, then
+            // add it to the split
+            auto deleteFilePath =
```

Comment:
This variable is not needed. The definition can be used in-place.

Reply:
Changing to:
```
IcebergDeleteFile deleteFile(
                FileContent::kPositionalDeletes,
                deleteFilePaths[deleteFileName].second->getPath(),
                fileFormat,
                deleteFilePaths[deleteFileName].first,
                testing::internal::GetFileSize(
                    std::fopen(deleteFilePath.c_str(), "r")));
```